### ISSUE TITLE
"too many open files" error in a large cluster

### ISSUE THREAD

**Issue by fgrzadkowski** [GoogleCloudPlatform/kubernetes#6351]
When I run a density test in a large cluster I've noticed the following error in scheduler and kubelet:

E0402 13:42:04.335868    4061 errors.go:66] apiserver received an error that is not an api.Status: Put http://127.0.0.1:4001/v2/keys/registry/pods/default/my-hostname-density30-07ce55c0-d93e-11e4-8f0b-bc30
5be02f04mlp5u?prevExist=false: dial tcp 127.0.0.1:4001: too many open files

This could be either because either etcd or apiserver has too low ulimit -n

**Comment by fgrzadkowski** [comment]
cc @wojtek-t

**Comment by fgrzadkowski** [comment]
Since etcd is running now in a pod it has a limit for fd set to 1000000 (10e6) so I'll see if increasing this for apiserver helps.

**Comment by ArtfulCoder** [comment]
etcd is running in a pod since yesterday afternoon.
(It was introduced first on friday evening, rolledback on sunday, and reintroduced yesterday afternoon.)

Just FYI, to help co-relate timelines..

**Comment by fgrzadkowski** [comment]
I increased the limit for apiserver to 10e6 and started counting opened fd using a simple bash loop:

```while true; do echo `date` : `ls /proc/4048/fd | wc -l`; sleep 0.2; done```

where 4048 is a PID of apiserver. I haven't noticed any more "too many" errors in few consecutive runs of density test on 50 node cluster with 30 pods per node. At the same time I've noticed that number of opened fd reached ~2000.

I'm gonna send a PR to increase the limit to something more reasonable.
